bam_helpers.py

```diff
@@ -280,51 +252,83 @@ def get_query_sequences(pileupcolumn, bam_out,
     errors = 0
 
     # add the reference position and the number
-    # of reads that correspond to this position
     temp.append(pileupcolumn.reference_pos)
-    temp.append(pileupcolumn.n)
 
-    try:
-        # append bases
-        temp.append(pileupcolumn.get_query_sequences(add_indels=use_indels))
-        bam_out.append(temp)
-    except Exception as e:
+    # if the count is zero then we consult the reference
+    # at this position if we have a reference file
+    if pileupcolumn.n == 0 and "fastadata" in kwargs:
 
-        # try a fall out extra step only if it makes sense
-        if do_not_use_indels_on_error and use_indels:
-            try:
+      # is this correct? 1 read only
+      temp.append(1)
```

Comment:
why add 1 here?

Reply:
ah since we consult the reference then it should be zero